trigger.tex

```diff
@@ -39,9 +40,84 @@ \chapter{Trigger Module}
     bits of \RcsrTselect but fewer than $2^m$ triggers.
 \end{commentary}
 
-It is possible for a trigger with the ``enter Debug Mode'' action (1) and another
-trigger with the ``raise a breakpoint exception'' action (0) to fire at the same
-time. The preferred behavior is to have both actions take place.  It is
+\section{Actions}
+
+Triggers can be configured to take one of several actions when they fire.
+Table~\ref{tab:action} lists all options.
+
+\begin{table}[H]
+\centering
+\caption{\FcsrMcontrolAction encoding}
+\label{tab:action}
+\begin{tabular}{|r|L|}
+\hline
+Value & Description \\
+\hline
+0 & Raise a breakpoint exception into M-Mode. (Used when software wants to
+    use the trigger module without an external debugger attached.)
+    If the trigger is precise, \Rmepc must contain the address of the
+    instruction that caused the trigger to fire. If the trigger is imprecise
+    \Rmepc will contain the address of the next instruction to be executed. \\
+\hline
+1 & Enter Debug Mode. (Only supported when the trigger's \FcsrTdataOneDmode
+    is 1.)
+    If the trigger is precise, \RcsrDpc must contain the address of the
+    instruction that caused the trigger to fire. If the trigger is imprecise
+    \RcsrDpc will contain the address of the next instruction to be executed. \\
+\hline
+2 -- 5 & Reserved for use by the trace specification. \\
+\hline
+other & Reserved for future use. \\
+\hline
+\end{tabular}
+\end{table}
+
+\section{Priority}
+
+Table~\ref{tab:priority} lists the synchronous exceptions from the Privileged
+Spec, and where the various types of triggers fit in. The first 3 columns come
+from the Privileged Spec, and the final column shows where triggers fit in. If
+this table contradicts the table in the Privileged Spec, then the latter takes
+precedence. This table only applies if triggers are precise. Otherwise triggers
+will fire some indeterminate time after the event, and the priority is
+irrelevant.
+
+\begin{table}[H]
+\centering
+\label{tab:priority}
+\begin{tabular}{|l|r|l|l|}
+  \hline
+  Priority      & Exception & Description & Trigger \\
+                &      Code &             & \\
+  \hline
+  {\em Highest} &          3 & & icount \\ \hline
+                &          3 & Instruction address breakpoint & dmcontrol address before execute \\ \hline
```

Comment:
I think you mean mcontrol (trigger type 2) instead of dmcontrol.  And should this table also include the itrigger/etrigger types?

Reply:
I do mean mcontrol. Thanks.

I decided not to include itrigger/etrigger because their description is already very clear when the trigger is taken. ("When the trigger fires, all CSRs are updated as defined by the Privileged Spec, and the requested action is taken just before the first instruction of the interrupt/exception handler is executed.") Furthermore interrupts aren't precise so don't belong in this table regardless. Including exceptions would clutter up the table a lot, and I don't think it would add clarity.

I'm happy to change things if that doesn't make sense.